src/main/java/org/quiltmc/loader/api/ModInitializer.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2016 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.loader.api;
+
+/**
+ * A mod initializer.
+ *
+ * <p>In {@code quilt.mod.json}, the entrypoint is defined with {@code main} key.</p>
+ *
+ * @see ClientModInitializer
+ * @see DedicatedServerModInitializer
+ * @see org.quiltmc.loader.api.QuiltLoader#getEntrypointContainers(String, Class)
+ */
+@FunctionalInterface
+public interface ModInitializer {
```

Comment:
These initializers (except for pre launch) should be moved to a minecraft package because not every game or application has a client, server and common mode.

Reply:
Slight correction: I've already placed client&server entrypoints in the `org.quiltmc.api.minecraft` package - is that package name ok?

My assumption is that *every* game which runs quilt mods would have some stage to perform initialisation of all mods - even if it's location is different for every game. As such a single `common` initialiser would make sense for cross-game mods, even if these are fairly uncommon. (But that's technically a whole different question - can you even make cross-game mods?)

Reply:
I guess we can look into that possibility later. For now I'll just move `ModInitialiser` to the minecraft package.

Reply:
Moved in https://github.com/QuiltMC/quilt-loader/pull/48/commits/fa0aa003da99ea7fbbf2149748126e9e4e9204d2